src/hdmf/spec/namespace.py

```diff
@@ -83,13 +85,16 @@ def contact(self):
 
     @property
     def author(self):
-        """String or list of strings with the authors or  None"""
+        """String or list of strings with the authors or None"""
         return self.get('author', None)
 
     @property
     def version(self):
-        """String, list, or tuple with the version or None """
-        return self.get('version', None)
+        """String, list, or tuple with the version or SpecNamespace.UNVERSIONED """
+        version = self.get('version', None)
+        if version is None:
+            self['version'] = SpecNamespace.UNVERSIONED
+        return self.get('version')
```

Comment:
`return self.get('version', None) or SpecNamespace.UNVERSIONED` probably do the same if version can't be numeric ;-) could also benefit of it is an empty list or an empty string for some reason

Reply:
The code as written also sets self['version']. How about changing line 95 to:
```python
if not version:
```

Reply:
I love findings with side effects, they are so much fun to debug ;-)
If anything, could also just not use version variable - used only once

Reply:
True, side effects are not great. I'll go with your suggestion.